linkml/generators/pydanticgen/pydanticgen.py

```diff
@@ -229,13 +247,71 @@ class PydanticGenerator(OOCodeGenerator):
     
     See :class:`.MetadataMode` for mode documentation
     """
+    split: bool = False
+    """
+    Generate schema that import other schema as separate python modules
+    that import from one another, rather than rolling all into a single
+    module (default, ``False``).
+    """
+    split_pattern: str = ".{{ schema.name }}"
+    """
+    When splitting generation, imported modules need to be generated separately
+    and placed in a python package and import from each other. Since the 
+    location of those imported modules is variable -- e.g. one might want to
+    generate schema in multiple packages depending on their version -- this
+    pattern is used to generate the module portion of the import statement.
+    
+    These patterns should generally yield a relative module import, 
+    since functions like :func:`.generate_split` will generate and write files
+    relative to some base file, though this is not a requirement since custom
+    split generation logic is also allowed.
+    
+    The pattern is a jinja template string that is given the ``SchemaDefinition``
+    of the imported schema in the environment. Additional variables can be passed
+    into the jinja environment with the ``split_environment`` argument.
```

Comment:
Should `split_environment` be `split_context` instead?

Reply:
Oh yep!

Reply:
done in https://github.com/linkml/linkml/pull/2208/commits/7fed49b3764d3f8c1ee84422444e675d0baec99c